Add JSON output to the verify command

dkovalev1 opened the pull request:
pgbackrest verify command result code is hard to relay on when it is required to
understand the correctness of the backup data in the repository. It is possible
to get more information by adding options --output text --verbose,

But parsing such output is hard and error prone. This patch implements
json format as a new output format for verify command which transforms the
output into json.

dkovalev1 committed:
verify json implementation

dkovalev1 committed:
Fix tests

dkovalev1 committed:
Merge branch 'integration' into ADBM-1976

dwsteele commented:
This is on my radar and I think it is a good change but it will be a few weeks or more before I can get to it. You hit us with a bunch of patches at once (which is great!) but we'll need to work through them one by one.

dkovalev1 committed:
Merge branch 'integration' into ADBM-1976

dkovalev1 committed:
Merge branch 'integration' into ADBM-1976

dkovalev1 commented:
Would it be possible to get an update on whether this PR is still under consideration?

dwsteele reviewed (changes requested):
I do think we want to add JSON to the verify command but I don't think this implementation is the right direction.

1) It should operate like the info command, i.e. text and JSON should be output to stdout separate from logging.
2) I don't think we should be constructing JSON manually here. It would be better to use JsonWrite to create the JSON.
3) Getting the format right here is important. I'm not convinced that what you have here is flexible enough, but I don't want to spend time on it unless the first two points are addressed.

dkovalev1 committed:
Merge branch 'pgbackrest:main' into ADBM-1976

dkovalev1 committed:
Implement JSON output in verify command (#82)

Implement JSON output in verify command

pgbackrest verify command result code is hard to rely on when it is required to
understand the correctness of the backup data in the repository. It is possible
to get more information by adding options --output text --verbose,

But parsing such output is hard and error prone. This patch implements
json format as a new output format for verify command which transforms the
output into json.

dkovalev1 commented:
Hi David,

I've made verify to operate like info and collect all the information into keyValue and then output this keyValue into text or JSON using JsonWrite. The resulting JSON is looks like this:
```
{
    "archives":[
        {
            "archiveId":"9.6-1",
            "checked":1,
            "checksumInvalid":0,
            "fileErrors":{"checksumInvalid":0,"missing":1,"other":0,"sizeInvalid":0},
            "missing":1,
            "other":0,
            "sizeInvalid":0,
            "valid":0
        }
    ],
    "backups":[
        {
            "checked":1,
            "checksumInvalid":0,
            "fileErrors":{"checksumInvalid":0,"missing":1,"other":0,"sizeInvalid":0},
            "label":"test-backup-label",
            "missing":1,
            "other":0,
            "sizeInvalid":0,
            "status":"invalid",
            "valid":0
        }
    ],
    "messages":[
        {"level":5,"message":"archiveId: 9.6-1, wal start: 0, wal stop: 2"}
    ]
}
```

dkovalev1 committed:
Remove usused strEscape function

dkovalev1 commented:
Thanks, David, will be waiting for your feedback. Meanwhile I've removed strEscape().

dkovalev1 commented:
Hi David! Have you got any thoughts regarding verify and JSON?

dwsteele commented:
For the time being I don't want to consider any large changes in this module. I have some refactors and improvements planned first.
